src/codeu/chat/DerbyStore.java

```diff
@@ -34,90 +57,155 @@ public DerbyStore() {
+			}
+	
+			// Connect to the database while creating a new schema
 			conn = DriverManager.getConnection(protocol + "chatapp;create=true", null);
 			
 			// Create a statement object to send queries
 			stmt = conn.createStatement();
 			
-			// If this table exists then we just continue
-			// since we don't want to create tables that exist
-			stmt.execute("SELECT * FROM chatuser");
+			// Create the chat user table
+			stmt.execute("CREATE TABLE " + userTableName + "(id varchar(255), name varchar(255), password varchar(255), creation BIGINT)");
+			
+			// Create the conversations table
+			stmt.execute("CREATE TABLE " + conversationTableName + "(id varchar(255), "
+					+ "owner varchar(255), creation BIGINT, title varchar(255), users varchar(255), firstMessage varchar(255)," +
+			"lastMessage varchar(255))");
+			
+			// Create the message table
+			stmt.execute("CREATE TABLE " + messageTableName + "(id varchar(255),"
+					 + "previous varchar(255), creation BIGINT, author varchar(255), content varchar(255), nextMessage varchar(255))");
+			
+			stmt.execute("CREATE TABLE " + chatParticipantsTableName + "(conversationid varchar(255), userid varchar(255))");
 			
-			// Give confirmation of connection
-			System.out.println("Tables exist. Connection made.");
+			// Give confirmation of execution.
+			LOG.info("Tables do not exist. Table creation executed.");
 		}
 		catch (Exception ex) {
-						try {
-						// Create the chat user table
-						stmt.execute("CREATE TABLE chatuser(id varchar(255), name varchar(255), password varchar(255), creation BIGINT)");
-						
-						// Create the conversations table
-						stmt.execute("CREATE TABLE conversation(id varchar(255), "
-								+ "owner varchar(255), creation BIGINT, title varchar(255), users varchar(255), firstMessage varchar(255)," +
-						"lastMessage varchar(255))");
-						
-						// Create the message table
-						stmt.execute("CREATE TABLE message(id varchar(255),"
-								 + "previous varchar(255), creation BIGINT, author varchar(255), content varchar(255), nextMessage varchar(255))");
-						
-						// Give confirmation of execution.
-						System.out.println("Tables do not exist. Table creation executed.");
-						} catch (Exception e) {
-							e.printStackTrace();
-						}
 			ex.printStackTrace();
 		}
 	}
 	
-	public boolean checkForUsername(String username) throws SQLException {
-		return stmt.execute("SELECT * FROM chatuser WHERE password = '" + username + "'" );
+	public boolean checkUsernameExists(String name) throws SQLException {
+		PreparedStatement checkUserTest = conn.prepareStatement(checkUsername);
+		checkUserTest.setString(1, name);
+		ResultSet user = checkUserTest.executeQuery();
+		
+		return user == null;
 	}
 	
-	public boolean checkForPassword(String password) throws SQLException {
-		return stmt.execute("SELECT * FROM chatuser WHERE password = '" + password + "'" );
+	public String userLoginCheck(String name, String password) throws SQLException {
```

Comment:
What is this method checking about the user login? You might want to rename this method or add a comment that explains what it does.